### PR TITLE
Added webdav tests

### LINKED ISSUE #86: Tests for WebDAV storage backend
Add tests for WebDAV storage backend.

### PR DESCRIPTION
**Pull request by craftamap**
Fixes #86.

### CODE REVIEW

File: admin/WEBDAV_CONFIG_IN [hunk 1]
```diff
@@ -0,0 +1,10 @@
+address: 0.0.0.0
```
Comment (muesli) [inline admin/WEBDAV_CONFIG_IN:1]: I'd call it `webdav/config.in`. (Note: `webdav/` being a sub-dir)

Comment (muesli) [reply admin/WEBDAV_CONFIG_IN:1]: Actually, if we're not replacing any values in the config file, we may just call it `webdav/config`. The `.in` suffix is commonly used when you're keeping placeholders in a file that will be dynamically replaced by the test/build.

Comment (craftamap) [reply admin/WEBDAV_CONFIG_IN:1]: do you mean `admin/webdav/config` or directly `webdav/config`?

Comment (muesli) [reply admin/WEBDAV_CONFIG_IN:1]: `admin/webdav/config` - it's arguably overhead for just one file, but I can already see tests for backends requiring multiple files on the horizon.

File: storage/webdav/webdav_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// +build backend
+
+package webdav
+
+/*
+ * knoxite
+ *     Copyright (c) 2016-2020, Christian Muehlhaeuser <muesli@gmail.com>
```
Comment (muesli) [inline storage/webdav/webdav_test.go:7]: Feel free to add your copyright.

**Pull request merged**